src/main/java/org/hobbit/core/components/AbstractEvaluationStorage.java

```diff
@@ -285,4 +211,275 @@ public void close() throws IOException {
+                 long timestamp = receiveTimeStamp ? buffer.getLong() : System.currentTimeMillis();
+                 receiveResponseData(taskId, timestamp, responseData);
+                 // If we should send acknowledgments (and there was no
+                 // error until now)
+                 if (ackChannel != null) {
+                     try {
+                         //ackChannel.basicPublish(ackExchangeName, "", null, RabbitMQUtils.writeString(taskId));
+                    	 ByteBuffer buf = ByteBuffer.wrap(RabbitMQUtils.writeString(taskId));
+                    	 ackChannel.writeBytes(buf, ackExchangeName, "", null);
+                     } catch (Exception e) {
+                         LOGGER.error("Error while sending acknowledgement.", e);
+                     }
+                     LOGGER.trace("Sent ack {}.", taskId);
+                 }
+            }
+        };
+    }
+    
+    private Object getSystemDirectHandler(boolean receiveTimeStamp, String ackExchangeName) {
+    	return new DirectCallback() {
+    		@Override
+			public void callback(byte[] data, List<Object> classs, BasicProperties props) {
+    			 ByteBuffer buffer = ByteBuffer.wrap(data);
+                 String taskId = RabbitMQUtils.readString(buffer);
+                 LOGGER.trace("Received from system {}.", taskId);
+                 byte[] responseData = RabbitMQUtils.readByteArray(buffer);
+                 long timestamp = receiveTimeStamp ? buffer.getLong() : System.currentTimeMillis();
+                 receiveResponseData(taskId, timestamp, responseData);
+                 // If we should send acknowledgments (and there was no
+                 // error until now)
+                 if (ackChannel != null) {
+                     try {
+                         //ackChannel.basicPublish(ackExchangeName, "", null, RabbitMQUtils.writeString(taskId));
+                    	 ByteBuffer buf = ByteBuffer.wrap(RabbitMQUtils.writeString(taskId));
+                    	 ackChannel.writeBytes(buf, ackExchangeName, null, null);
+                     } catch (Exception e) {
+                         LOGGER.error("Error while sending acknowledgement.", e);
+                     }
+                     LOGGER.trace("Sent ack {}.", taskId);
+                 }
+
+			}
+
+   		};
+    }
+    
+    private Object getConsumerCallback(String queueName) {
+    	Object consumerCallback = null;
+        if(isRabbitMQEnabled()) {
+        	consumerCallback = getDefaultConsumer(queueName);
+        } else {
+        	consumerCallback = getDirectConsumer(queueName);
+        }
+        return consumerCallback;
+    }
+    
+    private Object getDefaultConsumer(String queueName) {
+    	
+    	try {
+			evalModule2EvalStoreQueue = ((RabbitMQChannel)getFactoryForIncomingDataQueues()).getCmdQueueFactory()
+			        .createDefaultRabbitQueue(generateSessionQueueName(queueName));
+		} catch (IOException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
```

Comment:
What about error handling? Here, the wrong object will be returned while the method should return `null`.